packages/widget-v2/src/hooks/useGetSourceBalance.ts

```diff
@@ -15,6 +15,16 @@ export const useGetSourceBalance = () => {
     const { chainID, denom } = sourceAsset;
     if (!denom || !chainID) return;
 
+    const denomsExists = !!skipBalances?.chains?.[chainID]?.denoms;
+    const sourceBalance = skipBalances?.chains?.[chainID]?.denoms?.[denom];
+
+    if (denomsExists && sourceBalance === undefined) {
+      return {
+        amount: 0,
+        formattedAmount: "0",
+        error: undefined,
+      };
+    }
```

Comment:
this is because when we don't pass in any denoms, the server will not return back the denoms for the balances that are 0, but if `skipBalances?.chains?.[chainID]?.denoms` is found we can assume the server checked those denoms, but determined they were 0